File: src/config/mnemonics.js
// Preset mnemonic derivation profiles for Solana (ed25519/SLIP-0010)
// These capture common wallet defaults. Indices may vary by account.

export const MNEMONIC_PROFILES = {
  // Trust Wallet, Phantom and Solflare often use the non-change path at account 0
  trustwallet: { path: "m/44'/501'/0'" },
  phantom: { path: "m/44'/501'/0'" },
  solflare: { path: "m/44'/501'/0'" },
  // Solana CLI-compatible hardened path variant (frequently seen)
  solana_cli: { path: "m/44'/501'/0'/0'" },
};

export function getMnemonicProfile(name = 'trustwallet') {
  const key = String(name || '').toLowerCase();
  return MNEMONIC_PROFILES[key] || MNEMONIC_PROFILES.trustwallet;
}

export function listMnemonicProfiles() {
  return Object.keys(MNEMONIC_PROFILES);
}

File: src/config/mnemonics.rs
pub struct MnemonicProfile {
    pub name: &'static str,
    pub path: &'static str,
}

const MNEMONIC_PROFILES: &[MnemonicProfile] = &[
    MnemonicProfile {
        name: "trustwallet",
        path: "m/44'/501'/0'",
    },
    MnemonicProfile {
        name: "phantom",
        path: "m/44'/501'/0'",
    },
    MnemonicProfile {
        name: "solflare",
        path: "m/44'/501'/0'",
    },
    MnemonicProfile {
        name: "solana_cli",
        path: "m/44'/501'/0'/0'",
    },
];

pub fn get_mnemonic_profile(name: &str) -> &'static str {
    let key = name.trim().to_lowercase();
    MNEMONIC_PROFILES
        .iter()
        .find(|profile| profile.name == key)
        .map(|profile| profile.path)
        .unwrap_or(MNEMONIC_PROFILES[0].path)
}

pub fn list_mnemonic_profiles() -> Vec<&'static str> {
    MNEMONIC_PROFILES.iter().map(|p| p.name).collect()
}
